client/src/app/components/card-list.jsx

```diff
@@ -20,7 +20,7 @@ var CardList = React.createClass({
       }
     };
 
-    if (this.props.valid === false) {
+    if (this.props.valid === false || this.props.data.length < 1) {
```

Comment:
Checking length is more logical than using valid props. I'll remove valid props later commit.